### PR TITLE
Split kubefin.yaml into multiple components

### PR DESCRIPTION
**Pull request by REestrella**
Split kubefin.yaml into multiple components so that users can install the corresponding components based on their demand.

### CODE REVIEW

File: hack/release-yamls.sh [hunk 1]
```diff
@@ -28,8 +28,11 @@ rm -fr ${YAML_OUTPUT_DIR}/*.yaml
 hack/init-primary-config.sh auto cluster-0 true "${TAG}"
 
 # Generated KubeFin component YAML files
-readonly KUBEFIN_YAML=${YAML_OUTPUT_DIR}/kubefin.yaml
 readonly KUBEFIN_CRD_YAML=${YAML_OUTPUT_DIR}/kubefin_crd.yaml
+readonly KUBEFIN_AGENT_YAML=${YAML_OUTPUT_DIR}/kubefin-agent.yaml
+readonly KUBEFIN_COST_ANALYZER_YAML=${YAML_OUTPUT_DIR}/kubefin-cost-analyzer.yaml
+readonly KUBEFIN_GRAFANA_YAML=${YAML_OUTPUT_DIR}/kubefin-grafana.yaml
+readonly KUBEFIN_MIMIR_YAML=${YAML_OUTPUT_DIR}/kubefin-mimir.yaml
```
[the remark this answers is not in the document]
Comment (REestrella) [reply hack/release-yamls.sh:35]: Agreed.

File: hack/release-yamls.sh [hunk 2]
```diff
@@ -28,8 +28,9 @@ rm -fr ${YAML_OUTPUT_DIR}/*.yaml
 hack/init-primary-config.sh auto cluster-0 true "${TAG}"
 
 # Generated KubeFin component YAML files
-readonly KUBEFIN_YAML=${YAML_OUTPUT_DIR}/kubefin.yaml
 readonly KUBEFIN_CRD_YAML=${YAML_OUTPUT_DIR}/kubefin_crd.yaml
```
Comment (mrhello369) [inline hack/release-yamls.sh:31]: Should we also modify `kubefin-crd.yaml`?

**Review by mrhello369 — COMMENTED** [review]
Other lgtm

**Pull request merged**